frontend/model/contracts/group.js

```diff
@@ -336,6 +336,14 @@ DefineContract({
       process (state, { data, meta }) {
         var groupProfile = state.profiles[meta.username]
         const nonMonetary = groupProfile.nonMonetaryContributions
+
+        // Prevent stale amounts when the user changes their incomeDetailsType
+        if (data.incomeDetailsType === 'pledgeAmount') {
+          Vue.set(groupProfile, 'incomeAmount', null)
+        } else if (data.incomeDetailsType === 'incomeAmount') {
+          Vue.set(groupProfile, 'pledgeAmount', null)
+        }
+
```

Comment:
What is the reason for this logic? I have a few issues with it:

- It seems unnecessary
- The comment and code assume that we are changing `incomeDetailsType` in this action, which we might not be doing at all

Everything in this file needs to be as clean and minimal as possible, since it will all be "set in stone", and will define a protocol that alternative implementations will have to implement. So anything that isn't necessary should be removed.

Reply:
Removed, it's not necessary indeed. However, I still don't like the idea of leaving "traces" of stale data, there's no reason to keep a value that doesn't make sense anymore.

Reply:
You might be able to fix that in the UI portion of the app

Reply:
> You might be able to fix that in the UI portion of the app

hmm but there's nothing to fix ou the UI side. this is a state optimization, not UI bug.

Reply:
There is definitely something that can be fixed on the UI side: reset the value. The effect to the user will be the same, while the logic in our contract chains will be kept to a minimum

Reply:
I think I'm not understanding... reset the value where exactly? At the moment is all working fine.

Reply:
In the modal for the income details. If you're worried about a stale value appearing there, that's where you can reset it.